engine/gamesys/src/gamesys/scripts/script_physics.cpp

```diff
@@ -373,6 +391,13 @@ namespace dmGameSystem
         Vectormath::Aos::Point3 from( *dmScript::CheckVector3(L, 1) );
         Vectormath::Aos::Point3 to( *dmScript::CheckVector3(L, 2) );
 
+        if (Vectormath::Aos::lengthSqr(to - from) <= 0.0f)
+        {
+            dmLogOnceWarning("Ray had 0 length when ray casting, ignoring request.");
+            lua_pushnil(L);
+            return 1;
+        }
```

Comment:
I moved this check from the c++ libraries, to here, since that's essentially where it's used from.